src/luaclang.c

```diff
@@ -1,14 +1,367 @@
+{
+        CXIndex *idx = (CXIndex*) luaL_checkudata(L, n, INDEX_METATABLE);
+        return *idx;
+}
+
+/* Create CXTranslatioUnit as userdata */
+
+static CXTranslationUnit * new_CXTU(lua_State *L)
+{
+        CXTranslationUnit *tu = (CXTranslationUnit*) lua_newuserdata(L, sizeof(CXTranslationUnit));
+        luaL_getmetatable(L, TU_METATABLE);
+        lua_setmetatable(L, -2);
+        return tu;
+}
+
+/* Convert userdata type to CXTranslatioUnit */
+
+static CXTranslationUnit to_CXTU(lua_State *L, int n) 
+{
+        CXTranslationUnit *tu = (CXTranslationUnit*) luaL_checkudata(L, n, TU_METATABLE);
+        return *tu;
+}
+
+/* Create CXCursor as userdata */
+
+static CXCursor * new_CXCursor(lua_State *L) 
+{
+        CXCursor *cur = (CXCursor*) lua_newuserdata(L, sizeof(CXCursor));
+        luaL_getmetatable(L, CURSOR_METATABLE);
+        lua_setmetatable(L, -2);
+        return cur;
+}
+
+/* Convert userdata type to CXCursor */
+
+static CXCursor to_CXCursor(lua_State *L, int n) 
+{
+        CXCursor *c = (CXCursor*) luaL_checkudata(L, n, CURSOR_METATABLE);
+        return *c;
+}
+
+/* Create CXType as userdata */
+
+static CXType * new_CXType(lua_State *L)
+{
+        CXType *t = (CXType*) lua_newuserdata(L, sizeof(CXType));
+        luaL_getmetatable(L, TYPE_METATABLE);
+        lua_setmetatable(L, -2);
+        return t;
+}
+
+
+/* --Clang function-- */
+
+/*      
+        Format - createIndex(exclude_pch, diagnostics)
+        Parameters - 1. exclude_pch - When non-zero, allows enumeration of "local" declarations
+                     2. diagnostics - Display diagnostics
+        More info - https://clang.llvm.org/doxygen/group__CINDEX.html#ga51eb9b38c18743bf2d824c6230e61f93
+*/
+static int create_CXIndex(lua_State *L) 
+{
+        int exclude_pch = lua_toboolean(L, 1);
+        int diagnostics = lua_toboolean(L, 2);
+        CXIndex *idx = new_CXIndex(L);
+        *idx = clang_createIndex(exclude_pch, diagnostics);
+        return 1;
+}
+
+static luaL_Reg clang_function[] = {
+        {"createIndex", create_CXIndex},
+        {NULL, NULL}
+};
+
+
+/* --Index functions-- */
+
+/*      
+        Format - idx:disposeIndex()
+        Parameter -  idx - Index to be destroyed       
+        More info - https://clang.llvm.org/doxygen/group__CINDEX.html#ga166ab73b14be73cbdcae14d62dbab22a
+*/
+static int dispose_CXIndex(lua_State *L) 
+{
+        CXIndex idx = to_CXIndex(L, 1);
+        luaL_argcheck(L, idx != NULL, 1, "arg disposed");
+        clang_disposeIndex(idx);
+        idx = NULL;
+        return 0;
+}
+
+/*      
+        Format - idx:parseTU(file_name)
+        Parameters - 1. idx - The index object with which the translation unit will be associated  
+                     2. file_name - The name of the source file to load    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#ga2baf83f8c3299788234c8bce55e4472e
+*/
+static int parse_TU(lua_State *L)
```

Comment:
We could create a single Lua object that contains both the index and the translationunit to simplify the Lua API.
To create this object, you need to declare a struct with both:
```
struct clangparser {
  CXIndex idx;
  CXTranslationUnit tu;
};
```
Then, create a single function to initialise this object.